runtime/lua/vim/treesitter/query.lua

```diff
@@ -925,9 +919,9 @@ end
 ---   - max_start_depth (integer) if non-zero, sets the maximum start depth
 ---     for each match. This is used to prevent traversing too deep into a tree.
 ---   - match_limit (integer) Set the maximum number of in-progress matches (Default: 256).
----   - all (boolean) When set, the returned match table maps capture IDs to a list of nodes.
----     Older versions of iter_matches incorrectly mapped capture IDs to a single node, which is
----     incorrect behavior. This option will eventually become the default and removed.
+---   - all (boolean) When set (the default), the returned match table maps capture IDs to a list of
+---     nodes. Older versions of iter_matches incorrectly mapped capture IDs to a single node, which
+---     is incorrect behavior. This option will be removed in a future release.
```

Comment:
```suggestion
--- - all (boolean) When `false` (default `true`), the returned table maps capture IDs to a single
---   (last) node instead of the full list of matching nodes. This option is only for backward 
---   compatibility and will be removed in a future release.
```

Reply:
Changed as suggested.